src/ViewLayer/RaceModeDisplay/RaceModeDashboardView.cpp

```diff
@@ -179,11 +186,19 @@ void RaceModeDashboardView::connectMotorFaults(MotorFaultsPresenter& motorFaults
             this, SLOT(motorOneLimitFlagsReceived(LimitFlags)));
 }
 
+
 void RaceModeDashboardView::updateFaultLabel(QLabel& dashboardLabel, FaultLabel faultLabel)
 {
     if (faultLabel.priority() >= 0)
     {
         dashboardLabel.setStyleSheet(QString("color:%1").arg(faultLabel.color().name()));
+
+        if (backgroundAnimation_->currentTime() == 0 && dashboardLabel.text() != faultLabel.text())
```

Comment:
To be more explicit about the desired behavior I think it would be better if the 'if' statement here was something along the lines of `if(faultsList_.numberOfActiveFaultLabels >0)`
In this case, managing the animation should be in it's own function/slot that is called when the different faults are updated.